collectors/python.d.plugin/energid/energid.chart.py

```diff
@@ -0,0 +1,150 @@
+        'lines': [
+            ['blockchain_difficulty', 'Diff', 'absolute'],
+        ],
+    },
+    'mempool': {
+        'options': [None, 'MemPool', 'MiB', 'memory', 'energid.mempool', 'area'],
+        'lines': [
+            ['mempool_max', 'Max', 'absolute', None, 1024*1024],
+            ['mempool_current', 'Usage', 'absolute', None, 1024*1024],
+            ['mempool_txsize', 'TX Size', 'absolute', None, 1024*1024],
+        ],
+    },
+    'secmem': {
+        'options': [None, 'Secure Memory', 'KiB', 'memory', 'energid.secmem', 'area'],
+        'lines': [
+            ['secmem_total', 'Total', 'absolute', None, 1024],
+            ['secmem_locked', 'Locked', 'absolute', None, 1024],
+            ['secmem_used', 'Used', 'absolute', None, 1024],
+        ],
+    },
+    'network': {
+        'options': [None, 'Network', 'count', 'network', 'energid.network', 'line'],
+        'lines': [
+            ['network_connections', 'Connections', 'absolute'],
+        ],
+    },
+    'timeoffset': {
+        'options': [None, 'Network', 'seconds', 'network', 'energid.timeoffset', 'line'],
+        'lines': [
+            ['network_timeoffset', 'offseet', 'absolute'],
+        ],
+    },
+    'utxo': {
+        'options': [None, 'UTXO', 'count', 'UTXO', 'energid.utxo', 'line'],
+        'lines': [
+            ['utxo_count', 'UTXO', 'absolute'],
+        ],
+    },
+    'xfers': {
+        'options': [None, 'UTXO', 'count', 'UTXO', 'energid.xfers', 'line'],
+        'lines': [
+            ['utxo_xfers', 'Xfers', 'absolute'],
+        ],
+    },
+}
+
+METHODS = {
+    'getblockchaininfo': lambda r: {
+        'blockchain_blocks': r['blocks'],
+        'blockchain_headers': r['headers'],
+        'blockchain_difficulty': r['difficulty'],
+    },
+    'getmempoolinfo': lambda r: {
+        'mempool_txcount': r['size'],
+        'mempool_txsize': r['bytes'],
+        'mempool_current': r['usage'],
+        'mempool_max': r['maxmempool'],
+    },
+    'getmemoryinfo': lambda r: dict([
+        ('secmem_' + k, v) for (k,v) in r['locked'].items()
+    ]),
+    'getnetworkinfo': lambda r: {
+        'network_timeoffset' : r['timeoffset'],
+        'network_connections': r['connections'],
+    },
+    'gettxoutsetinfo': lambda r: {
+        'utxo_count' : r['txouts'],
+        'utxo_xfers' : r['transactions'],
+        'utxo_size' : r['disk_size'],
+        'utxo_amount' : r['total_amount'],
+    },
+}
+
+class Service(UrlService):
+    def __init__(self, configuration=None, name=None):
+        UrlService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.host = self.configuration.get('host', '127.0.0.1')
+        self.port = self.configuration.get('port', 9796)
+        self.url = '{scheme}://{host}:{port}'.format(
+            scheme=self.configuration.get('scheme', 'http'),
+            host=self.host,
+            port=self.port,
+        )
```

Comment:
Why do we need host, port and scheme? i suggest to use `url` (es is a bad example :smile: )

Reply:
In general, only `port` configuration is needed. There is no URL involved in most other related configuration files of the daemon itself and dependent services.

Reply:
well, so you mean host is always localhost. Still `url` will do. And it is 1 param instead of 3. I suggest to use it.

Reply:
In most cases, the host is '127.0.0.1' and not even localhost which may resolve to IPv6 ::1.

Reply:
The main point here, it is NOT HTTP service. It is RPC which is framed with bare simple HTTP.